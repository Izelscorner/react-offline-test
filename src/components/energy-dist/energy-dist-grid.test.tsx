import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import EnergyDistGrid from "./energy-dist-grid";
import pretty from "pretty";

describe("<EnergyDistGrid />", () => {
  const data = {
    generationmix: [
      { fuel: "biomass", perc: 20 },
      { fuel: "nuclear", perc: 40 },
      { fuel: "wind", perc: 40 }
    ]
  };
  let fetchData: () => any = jest.fn(
    async () => await Promise.resolve({ data })
  );
  let fetchDataError: () => any = jest.fn(
    async () => await Promise.reject("error")
  );

  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("matches snapshot", async () => {
    await act(async () => {
      render(<EnergyDistGrid getData={fetchData} />, container);
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it("3 colums found after data fetch", async () => {
    await act(async () => {
      render(<EnergyDistGrid getData={fetchData} />, container);
    });
    expect(container.querySelectorAll(".col-3").length).toBe(3);
  });

  it("Loading text disappears after data request.", async () => {
    await act(async () => {
      render(<EnergyDistGrid getData={fetchData} />, container);
    });
    expect(container.querySelectorAll(".isLoading").length).toBe(0);
  });

  it("Error text after failed data fetch.", async () => {
    await act(async () => {
      render(<EnergyDistGrid getData={fetchDataError} />, container);
    });
    expect(container.querySelectorAll(".error").length).toBe(1);
  });

});
