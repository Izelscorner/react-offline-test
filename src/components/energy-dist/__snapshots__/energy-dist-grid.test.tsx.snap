// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnergyDistGrid /> matches snapshot 1`] = `
"<div class=\\"row\\">
  <div class=\\"time center\\">01:00:00 GMT+0100 (BST) - 01:30:00 GMT+0100 (BST)</div>
  <div class=\\"col-3\\">
    <div class=\\"tile-card center\\">
      <div class=\\"tile-body\\"><label class=\\"tile-label\\">biomass</label>
        <div id=\\"gauge-chart-biomass-0\\" style=\\"width: 100%;\\"><svg width=\\"0\\" height=\\"0\\">
            <g transform=\\"translate(0, 0)\\">
              <g class=\\"doughnut\\" transform=\\"translate(0, 0)\\">
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 95, 109);\\"></path>
                </g>
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 141, 104);\\"></path>
                </g>
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 195, 113);\\"></path>
                </g>
              </g>
              <g class=\\"needle\\" transform=\\"translate(0, 0)\\">
                <path d=\\"M 0 0 L 0 0 L 0 0\\" fill=\\"#000\\"></path>
                <circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"#464A4F\\"></circle>
              </g>
            </g>
          </svg></div>
        <h5>20%</h5>
      </div>
    </div>
  </div>
  <div class=\\"col-3\\">
    <div class=\\"tile-card center\\">
      <div class=\\"tile-body\\"><label class=\\"tile-label\\">nuclear</label>
        <div id=\\"gauge-chart-nuclear-1\\" style=\\"width: 100%;\\"><svg width=\\"0\\" height=\\"0\\">
            <g transform=\\"translate(0, 0)\\">
              <g class=\\"doughnut\\" transform=\\"translate(0, 0)\\">
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 95, 109);\\"></path>
                </g>
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 141, 104);\\"></path>
                </g>
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 195, 113);\\"></path>
                </g>
              </g>
              <g class=\\"needle\\" transform=\\"translate(0, 0)\\">
                <path d=\\"M 0 0 L 0 0 L 0 0\\" fill=\\"#000\\"></path>
                <circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"#464A4F\\"></circle>
              </g>
            </g>
          </svg></div>
        <h5>40%</h5>
      </div>
    </div>
  </div>
  <div class=\\"col-3\\">
    <div class=\\"tile-card center\\">
      <div class=\\"tile-body\\"><label class=\\"tile-label\\">wind</label>
        <div id=\\"gauge-chart-wind-2\\" style=\\"width: 100%;\\"><svg width=\\"0\\" height=\\"0\\">
            <g transform=\\"translate(0, 0)\\">
              <g class=\\"doughnut\\" transform=\\"translate(0, 0)\\">
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 95, 109);\\"></path>
                </g>
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 141, 104);\\"></path>
                </g>
                <g class=\\"arc\\">
                  <path d=\\"M0,0Z\\" style=\\"fill: rgb(255, 195, 113);\\"></path>
                </g>
              </g>
              <g class=\\"needle\\" transform=\\"translate(0, 0)\\">
                <path d=\\"M 0 0 L 0 0 L 0 0\\" fill=\\"#000\\"></path>
                <circle cx=\\"0\\" cy=\\"0\\" r=\\"0\\" fill=\\"#464A4F\\"></circle>
              </g>
            </g>
          </svg></div>
        <h5>40%</h5>
      </div>
    </div>
  </div>
</div>"
`;
